xds/type/matcher/v3/http_inputs.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package xds.type.matcher.v3;
+
+import "xds/annotations/v3/status.proto";
+
+import "validate/validate.proto";
+
+option java_package = "com.github.xds.type.matcher.v3";
+option java_outer_classname = "HttpInputsProto";
+option java_multiple_files = true;
+option go_package = "github.com/cncf/xds/go/xds/type/matcher/v3";
+
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#not-implemented-hide:]
+
+// [#protodoc-title: Common HTTP Inputs]
+
+// Match input indicates that matching is done on the set of :ref:`HTTP attributes
+// <https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes>`.
+// The attributes will be exposed via Common Expression Language runtime to associated CEL matcher,
+// or via ``get_property`` ABI method to associated Wasm extension based matcher.
+message HttpCelMatchInput {
+  // [#next-free-field: 7]
+  message Attributes {
```

Comment:
Can you explain the intent behind limiting visibility? Is it to "hide" subset of attributes and for what purpose?

I'd imagine that the HTTP input would try to provide all information to the extent possible and leave some values undefined if they are not available. E.g. response would be empty on request headers.

Reply:
CEL distinguishes between empty and missing values. It's an error if some top-level property is not defined which propagates as the evaluation result.

I think the presence of top-level attributes is pretty clear depending on the context. So the new capability is that one might want to hide some attributes when they are available. Is that capability really needed?

Reply:
Updated `HttpCelMatchInput` in https://github.com/envoyproxy/envoy/commit/8c20ec1384444fefd433a3b839ee0bc2f78bfed5. Now looking at `CelMatcher`.